### PR TITLE
feat(core): runtime, keep skipReason up to date

### PR DESCRIPTION
**Pull request by venetrius**
camunda/camunda-bpm-platform/issues/5439

## Related issues

closes #

### CODE REVIEW

File: core/src/main/java/io/camunda/migrator/impl/clients/DbClient.java [hunk 1]
```diff
@@ -93,6 +94,16 @@ public void updateC8KeyByC7IdAndType(String c7Id, Long c8Key, TYPE type) {
...
   }
 
+  public void updateSkipReason(String c7Id, TYPE type, String skipReason) {
+    if(!properties.getSaveSkipReason()) {
+      return;
+    }
+
+    DbClientLogs.updatingSkipReason(c7Id, skipReason);
+    var model = createIdKeyDbModel(c7Id, null, null, type, skipReason);
+    callApi(() -> idKeyMapper.updateSkipReason(model),FAILED_TO_UPDATE_SKIP_REASON + c7Id);
```
Comment (yanavasileva) [inline core/src/main/java/io/camunda/migrator/impl/clients/DbClient.java:104]: 🙃 formatting
```suggestion
    if (!properties.getSaveSkipReason()) {
      return;
    }

    DbClientLogs.updatingSkipReason(c7Id, skipReason);
    var model = createIdKeyDbModel(c7Id, null, null, type, skipReason);
    callApi(() -> idKeyMapper.updateSkipReason(model), FAILED_TO_UPDATE_SKIP_REASON + c7Id);
```

File: qa/src/test/java/io/camunda/migrator/qa/runtime/SkipReasonUpdateOnRetryTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    migratorProperties.setSaveSkipReason(true);
+
+    // given: deploy process with message start event only in C7
+    deployer.deployCamunda7Process("messageStartEventProcess.bpmn");
+    var process = runtimeService.startProcessInstanceByKey("MessageStartEventProcessId");
+
+    // when: run initial migration
+    runtimeMigrator.start();
+
+    // then: process instance should be skipped with no none start event error
```
Comment (yanavasileva) [inline qa/src/test/java/io/camunda/migrator/qa/runtime/SkipReasonUpdateOnRetryTest.java:79]: 🔧 wrong skip reason in the comment
```suggestion
    // then: process instance should be skipped with no C8 deployment error
```

File: qa/src/test/java/io/camunda/migrator/qa/runtime/SkipReasonUpdateOnRetryTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    deployer.deployCamunda8Process("messageStartEventProcess.bpmn");
+
+    // when: retry skipped process instances
+    runtimeMigrator.setMode(RETRY_SKIPPED);
+    runtimeMigrator.start();
+
+    // then: process instance should still be skipped but with updated reason (missing none start event)
+    String updatedSkipReason = String.format(NO_NONE_START_EVENT_ERROR, "MessageStartEventProcessId", 1);
+    verifySkipped(process, updatedSkipReason);
+//        "C8 process definition [id: MessageStartEventProcessId, version: 1] should have a None Start Event.");
```
Comment (yanavasileva) [inline qa/src/test/java/io/camunda/migrator/qa/runtime/SkipReasonUpdateOnRetryTest.java:93]: Do we need this here?
```suggestion
```

File: qa/src/test/java/io/camunda/migrator/qa/runtime/SkipReasonUpdateOnRetryTest.java
Comment (yanavasileva) [inline qa/src/test/java/io/camunda/migrator/qa/runtime/SkipReasonUpdateOnRetryTest.java:1]: ❓ Shall we add a test case for `saveSkipReason=false` that the reason is still null after a retry?

**Review by yanavasileva — CHANGES_REQUESTED** [review]
Looks good overall, small suggestions
